pkg/client/issuecredential/example_test.go

```diff
@@ -0,0 +1,369 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuecredential
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/messenger"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/issuecredential"
+	mocks "github.com/hyperledger/aries-framework-go/pkg/internal/gomocks/client/presentproof"
+	dispatchermocks "github.com/hyperledger/aries-framework-go/pkg/internal/gomocks/didcomm/dispatcher"
+	messengermocks "github.com/hyperledger/aries-framework-go/pkg/internal/gomocks/didcomm/messenger"
+	protocolmocks "github.com/hyperledger/aries-framework-go/pkg/internal/gomocks/didcomm/protocol/presentproof"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/mem"
+)
+
+// payload represents a transport message structure
+type payload struct {
+	msg      []byte
+	myDID    string
+	theirDID string
+}
+
+// nolint: gocyclo, gocognit
+func mockContext(agent string, tr map[string]chan payload) Provider {
+	ctrl := gomock.NewController(nil)
+
+	outbound := dispatchermocks.NewMockOutbound(ctrl)
+	outbound.EXPECT().
+		SendToDID(gomock.Any(), gomock.Any(), gomock.Any()).
+		DoAndReturn(func(msg interface{}, myDID, theirDID string) error {
+			src, err := json.Marshal(msg)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			tr[theirDID] <- payload{
+				msg:      src,
+				myDID:    theirDID,
+				theirDID: myDID,
+			}
+
+			return nil
+		}).AnyTimes()
+
+	store := mem.NewProvider()
+
+	mProvider := messengermocks.NewMockProvider(ctrl)
+	mProvider.EXPECT().StorageProvider().Return(store)
+	mProvider.EXPECT().OutboundDispatcher().Return(outbound)
+
+	msgSvc, err := messenger.NewMessenger(mProvider)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	provider := protocolmocks.NewMockProvider(ctrl)
+	provider.EXPECT().StorageProvider().Return(store)
+	provider.EXPECT().Messenger().Return(msgSvc)
+
+	svc, err := issuecredential.New(provider)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	go func() {
+		for {
+			select {
+			case msg := <-tr[agent]:
+				didMap, err := service.ParseDIDCommMsgMap(msg.msg)
+				if err != nil {
+					fmt.Println(err)
+				}
+
+				fmt.Println(agent, "received", didMap.Type(), "from", msg.theirDID)
+
+				if err = msgSvc.HandleInbound(didMap, msg.myDID, msg.theirDID); err != nil {
+					fmt.Println(err)
+				}
+
+				_, err = svc.HandleInbound(didMap, msg.myDID, msg.theirDID)
+				if err != nil {
+					fmt.Println(err)
+				}
+			case <-time.After(time.Second):
+				return
+			}
+		}
+	}()
+
+	mockProvider := mocks.NewMockProvider(ctrl)
+	mockProvider.EXPECT().Service(gomock.Any()).Return(svc, nil)
+
+	return mockProvider
+}
+
+// nolint: gocyclo
+func ExampleClient_SendOffer() {
+	transport := map[string]chan payload{
```

Comment:
Same comment I left you in the PR with the examples for presentproof... let's try to make these examples look more realistic next time.